core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -103,8 +103,14 @@ public RecordCursor cursor(
             getMaterializedViews(session, metadata, accessControl, tablePrefix).forEach((tableName, definition) -> {
                 QualifiedObjectName name = new QualifiedObjectName(tablePrefix.getCatalogName(), tableName.getSchemaName(), tableName.getTableName());
                 MaterializedViewFreshness freshness = metadata.getMaterializedViewFreshness(session, name);
-                Object[] materializedViewRow = createMaterializedViewRow(name, freshness, definition);
 
+                // Mitigates race condition where a materialized view is deleted
+                // after we retrieve it's info but before we retrieve its "freshness".
+                if (freshness.isNone()) {
```

Comment:
That's a weird SPI. If we want to encode optional result, `Optional<...>` is the right way to do it.
However, this is exceptionally optional result, so throwing something like `ViewNotFound` would be more appropriate.

cc @martint

Reply:
If we throw an exception, is there a reason to use a `MaterializedViewFreshness` class? Maybe a boolean is enough?